### PR TITLE
Assign emails

### PR DESCRIPTION
**Pull request by dmytsuu**

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -53,6 +53,14 @@ def user_agent?
     (headers && headers['User-Agent']).present?
   end
 
+  def occurrences_today?
+    occurrences.any? { |occurrence| occurrence.created_at.today? }
+  end
+
+  def occurrence_count_today
+    occurrences.select { |occurrence| occurrence.created_at.today? }.count
```
Comment (dmytsuu) [inline app/models/event.rb:61]: .where("created_at >= ?", Time.zone.now.beginning_of_day)

File: app/models/event.rb [hunk 2]
```diff
@@ -53,6 +53,14 @@ def user_agent?
     (headers && headers['User-Agent']).present?
   end
 
+  def occurrences_today?
+    occurrences.any? { |occurrence| occurrence.created_at.today? }
```
Comment (dmytsuu) [inline app/models/event.rb:57]: .where("created_at >= ?", Time.zone.now.beginning_of_day).count.positive?

File: app/services/events/update_service.rb [hunk 3]
```diff
@@ -15,6 +16,15 @@ def notify
     Integration.notify(event: event, action: action, reason: user.nickname)
   end
 
+  def notify_assignee?
+    assigne_user_id = declared_params.dig(:event, :user_id)
```
Comment (dmytsuu) [inline app/services/events/update_service.rb:20]: assignee_user_id

File: app/views/event_mailer/shared/_assigner_photo.slim [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.logo-wrapper
+  = link_to t('web_url.production') do
+    = image_tag(attachments['assigner.png'].url, class: 'bugno-logo rounded')
```
Comment (dmytsuu) [inline app/views/event_mailer/shared/_assigner_photo.slim:3]: rename bugno-logo class

File: app/mailers/event_mailer.rb [hunk 5]
```diff
@@ -24,8 +24,31 @@ def occurrence(occurence, addresses)
...
+    mail(to: @assignee.email,
+         subject: default_i18n_subject(project_name: @event.project.name,
+                                       environment: @event.environment,
+                                       title: @event.title, assigner_name: @assigner.nickname))
+  end
+
   private
 
+  def render_assignee_analytics?
+    @event.occurrence_count.positive? || @event.occurrences_today? || @assignee.assigned_in_project?(@event.project_id)
```
Comment (dmytsuu) [inline app/mailers/event_mailer.rb:43]: all these query methods reused in partial, should we consider putting them into instance variables?
@Alexey-Lukin

File: app/models/event.rb (hunk 1 above)
Comment (dmytsuu) [reply app/models/event.rb:61]: should be scope

File: app/models/event.rb [hunk 6]
```diff
@@ -53,6 +53,14 @@ def user_agent?
     (headers && headers['User-Agent']).present?
   end
 
+  def occurrences_today_count
+    occurrences.where('created_at >= ?', Time.zone.now.beginning_of_day).count
```
Comment (Alexey-Lukin) [inline app/models/event.rb:57]:
```
where('created_at >= ?', Time.zone.now.beginning_of_day)
```
maybe move into scope

File: app/views/event_mailer/occurrence.slim [hunk 7]
```diff
@@ -1,19 +1,8 @@
...
-        span.section-title> = t('mailer.file_path')
-        span = @chunk_of_code['filename']
-      .code-snippet = render partial: 'event_mailer/shared/chunk_of_code', object: @chunk_of_code, as: :line if @chunk_of_code
-  - if @event.backtrace.present?
-    .section
-      .section-title = t('mailer.backtrace')
-      .divider
-      .backtrace = render partial: 'event_mailer/shared/backtrace', collection: @event.backtrace[0..10], as: :line
+  = render partial: 'event_mailer/shared/chunk_of_code_section' if @chunk_of_code.present?
+  = render partial: 'event_mailer/shared/backtrace_section' if @event.backtrace.present?
```
Comment (Alexey-Lukin) [inline app/views/event_mailer/occurrence.slim:8]: please be aware that previous file and this one are almost the same

File: app/views/event_mailer/shared/_chunk_of_code_section.slim [hunk 8]
```diff
@@ -0,0 +1,7 @@
+.section
+  .section-title = t('mailer.code_snippet')
+  .divider
+  .filename
+    span.section-title> = t('mailer.file_path')
+    span = @chunk_of_code['filename']
```
Comment (Alexey-Lukin) [inline app/views/event_mailer/shared/_chunk_of_code_section.slim:6]: if @chunk_of_code will be nill you will have an exception

**Pull request merged**